.github/workflows/pull_request.yml

```diff
@@ -100,29 +86,10 @@ jobs:
           name: receptor
           path: /usr/local/bin/receptor
   receptorctl:
-    name: receptorctl
     needs: receptor
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v4
-        with:
-          fetch-depth: 0
-
-      - name: Download receptor
-        uses: actions/download-artifact@v4
-        with:
-          name: receptor
-          path: /usr/local/bin/
-
-      - name: Fix permissions on receptor binary
-        run: sudo chmod a+x /usr/local/bin/receptor
-
-      - name: Setup nox
-        uses: wntrblm/nox@2024.04.15
-
-      - name: Run receptorctl tests
-        run: make receptorctl-test
+    uses: ./.github/workflows/reusable_receptorctl_tests.yml
+  lint-receptorctl:
+    uses: ./.github/workflows/reusable_receptorctl_checks.yml
```

Comment:
So I've been staring at this for a while and couldn't understand why I had a feeling of a déjà vu.. I realized that I solved this much duplication in the past [elsewhere](https://github.com/aio-libs/frozenlist/blob/8c15ec9/.github/workflows/ci-cd.yml#L126-L157). I moved the matrixes out of the reusable workflow, which allowed me to have a single workflow called twice with different matrices. Here, it'd look as follows:
```suggestion
    strategy:
      fail-fast: false
      matrix:
        python-version:
        # NOTE: The highest and the lowest versions come
        # NOTE: first as their statuses are most likely to
        # NOTE: signal problems early:
        - 3.11
        - 3.8
        - >-
          3.10
        - 3.9
    uses: ./.github/workflows/reusable-nox.yml
    with:
      python-version: ${{ matrix.python-version }}
      session: tests-${{ matrix.python-version }}

  lint-receptorctl:
    strategy:
      fail-fast: false
      matrix:
        session:
        - check_style
        - check_format
    uses: ./.github/workflows/reusable-nox.yml
    with:
      python-version: 3.11
      session: ${{ matrix.session }}
```

And the, in `reusable-nox.yml`, you'd have to declare `inputs:` with entries for the Python version and session, remove the matrix definition from the workflow job, and reference the passed data via `${{ inputs }}`. Then, you could reuse this for the tests too, you'd just need to make normalization of how the inputs are used a bit more dynamic inside the reusable workflow.